snooty/specparser.py

```diff
@@ -439,15 +447,134 @@ def resolve_value(key: str, inheritable: _T) -> _T:
+                else defined_composable.default
+            )
+            default_option = next(
+                (
+                    option
+                    for option in merged_options
+                    if merged_default and option.id == merged_default
+                ),
+                None,
+            )
+            if merged_default and not default_option:
+                diagnostics.append(
+                    UnknownOptionId(
+                        "Spec composables default",
+                        merged_default,
+                        [option.title for option in merged_options],
+                        0,
+                    )
+                )
+            res.append(
+                Composable(
+                    defined_composable.id,
+                    merged_title,
+                    merged_default,
+                    merged_dependencies,
+                    merged_options,
+                )
+            )
+
+        for composableObj in custom_composable_by_id.values():
+            res.append(
+                Composable(
+                    composableObj["id"],
+                    composableObj["title"],
+                    composableObj["default"] if "default" in composableObj else None,
+                    (
+                        composableObj["dependencies"]
+                        if "dependencies" in composableObj
+                        else None
+                    ),
+                    list(
+                        map(
+                            lambda option: TabDefinition(option["id"], option["title"]),
+                            composableObj["options"],
+                        )
+                    ),
+                )
+            )
+
+        spec.composables = res
+        return (spec, diagnostics)
+
+    @classmethod
+    def initialize(cls, text: str, configPath: Optional[Path]) -> "Spec":
         cls.SPEC = Spec.loads(text)
+        if configPath:
+            project_config = tomli.loads(configPath.read_text(encoding="utf-8"))
+            # NOTE: would like to check_type but circular imports
+            # this is already verified earlier in the process
+            spec, diagnostics = cls._merge_composables(
+                cls.SPEC,
+                (
+                    project_config["composables"]
+                    if "composables" in project_config
+                    else []
+                ),
+            )
+            spec.merged = True
+            cls.SPEC = spec
+            for diagnostic in diagnostics:
+                logger.error(diagnostic)
+        return cls.SPEC
 
     @classmethod
-    def get(cls) -> "Spec":
-        if cls.SPEC is None:
-            path = util.PACKAGE_ROOT.joinpath("rstspec.toml")
-            cls.initialize(path.read_text(encoding="utf-8"))
-
-        spec = cls.SPEC
-        assert spec is not None
-        return spec
+    def get(cls, configPath: Optional[Path]) -> "Spec":
```

Comment:
(Nit) Since it seems like the most common use case is to call this with `configPath` set to `None`, it might be okay to have the default arg be `None`